react/components/withProductContext.js

```diff
@@ -0,0 +1,11 @@
+import React, { useContext } from 'react'
+import { ProductContext } from 'vtex.product-context'
+
+function withProductContext(Component) {
```

Comment:
this is a hook, it should not be called "with Product context", its breaking the pattern. A better name would be "useProductContext"

Reply:
That's a HOC abstracting a hook to a class component, it's ok